runtime/src/snapshot_utils.rs

```diff
@@ -471,6 +472,101 @@ pub fn move_and_async_delete_path(path: impl AsRef<Path> + Copy) {
         .unwrap();
 }
 
+pub fn clean_orphaned_account_snapshot_dirs(
```

Comment:
Now it returns a Result.  But I had to revert the unzip change to still have an intermediate vec of tuples, because from_iter cannot take a result from the closure.  not sure if there is a better way to do it.

Reply:
added the Result return.